### DISCUSSION TITLE
Amazon Q Developer: Keep history of previous chats

### DISCUSSION THREAD

**Discussion by tchristian32** [aws/aws-toolkit-vscode#5426] [Ideas]
It would be great if Amazon Q Developer was able to retain a history of chats so that if I come back in another time, I can jump back into a precious chat that I had. This is similar to how ChatGPT and others work where they keep the last 10 chats or something like that.

**Comment by animesh303** [comment]
Agree. VSCode can crash for any reason such as updates of extensions. For larger code implementation, if code exits before all steps are completed, I don't see how Amazon Q can possibly provide the same steps as earlier.

**Reply by nick-pateman-avalara** [reply]
Totally, I'm actually surprised this isn't a feature already.  You could spend hours getting what you want out of Amazon Q just to then lose all history of it. It's an essential feature. Even if historical items are in a "disconnected" state from the workspace, being able to look over it again should still work.

**Comment by palminha** [comment]
I'm pretty sure this was nicelly working previously... and don't understand when, but recently stopped working
